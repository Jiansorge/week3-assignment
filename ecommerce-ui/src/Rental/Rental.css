.rental{
    width:100%;
    /* border: 1px dashed red; */
}

li.rental {
    display: flex;
    flex-flow: column;
    padding:10px;
    /* margin: 8px; */
    color: black;
    text-decoration: none;
    align-items: flex-start;
    width: 50%;
    /* border: 1px solid red; */
}

.rental-info {
    display: flex;
    flex-flow: column;
    align-items: flex-start;
    width:100%;
    /* border: 1px solid blue; */
    text-align: left;

}

.rental-info .payment {
    display: flex;
    flex-flow: row wrap;
    justify-content: flex-start;
    width: 100%;
}

.image {
    width: 100%;
    object-fit: cover;
    height: 80%;
}

.title {
    text-transform: capitalize;
    padding: 0 0 4px;
    font-weight: 600;
    color: midnightblue;
}

.payment{
    padding: 0 0 4px;
    font-weight: 300;
    font-size: 0.9rem;
}