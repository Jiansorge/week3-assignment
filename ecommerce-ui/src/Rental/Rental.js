import React from 'react';
import Image from './components/Image.js';
import RentalDetail from './components/RentalDetail.js';
import Rating from './components/Rating.js';

import './Rental.css'
import './components/RentalDetail.css'
import './components/Rating.css'

import PropTypes from 'prop-types';

class Rental extends React.Component {
    static propTypes = {
        rental: PropTypes.object.isRequired,
        // key: PropTypes.number.isRequired,
        idx: PropTypes.number.isRequired,
        onClick: PropTypes.func.isRequired,
        buttonText: PropTypes.string.isRequired,
    }
    // const type = this.props.type;
    // console.log(this.props.type);
    render() {
        const onClick = this.props.onClick;

      return (
        <li className='rental' >
            <Image className="image" rental={this.props.rental}/>

            <div className="rental-info"
            >
                <RentalDetail rental={this.props.rental} />

                <h3 className='title'>
                    {this.props.rental.title}
                </h3>
                <div className="payment">
                    $ {this.props.rental.payment.cost} per night
                    { //Check if payment has description
                        (this.props.rental.payment.description !== "" )
                            ? <p> 
                                &nbsp; &#8226; &nbsp; {this.props.rental.payment.description} 
                              </p> 
                            : ""
                } </div>

                <Rating className="payment" rental={this.props.rental} />
                <button onClick={() => onClick(this.props.idx)}>
                    {this.props.buttonText}</button>
            </div>
        </li>
  
      )
    }
  }
 //
//   

  export default Rental;