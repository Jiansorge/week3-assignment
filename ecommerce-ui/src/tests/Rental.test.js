import React from 'react';
import Rental from '../Rental//Rental.js';
import {shallow} from 'enzyme';
import airbnbs from '../airbnbs.json';


//import testData from './testData.json';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<Image />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

it('should match the snapshot', () => {
  const wrapper = shallow(<Rental rental={airbnbs[0]} />);
  expect(wrapper).toMatchSnapshot();
});