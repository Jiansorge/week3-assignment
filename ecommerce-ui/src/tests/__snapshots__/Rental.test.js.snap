// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<li
  className="rental"
>
  <Image
    className="image"
    rental={
      Object {
        "host": Object {
          "isSuperhost": true,
          "name": "Robert",
        },
        "houseType": "Apartment",
        "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
        "location": Object {
          "city": "New York",
          "country": "USA",
        },
        "payment": Object {
          "cost": 105,
          "description": "Free cancellation",
        },
        "rating": Object {
          "reviews": 133,
          "stars": 3,
        },
        "title": "Centrally-located Manhattan studio",
      }
    }
  />
  <div
    className="rental-info"
  >
    <RentalDetail
      rental={
        Object {
          "host": Object {
            "isSuperhost": true,
            "name": "Robert",
          },
          "houseType": "Apartment",
          "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
          "location": Object {
            "city": "New York",
            "country": "USA",
          },
          "payment": Object {
            "cost": 105,
            "description": "Free cancellation",
          },
          "rating": Object {
            "reviews": 133,
            "stars": 3,
          },
          "title": "Centrally-located Manhattan studio",
        }
      }
    />
    <h3
      className="title"
    >
      Centrally-located Manhattan studio
    </h3>
    <div
      className="payment"
    >
      $ 
      105
       per night
      <p>
          •   
        Free cancellation
      </p>
       
    </div>
    <Rating
      className="payment"
      rental={
        Object {
          "host": Object {
            "isSuperhost": true,
            "name": "Robert",
          },
          "houseType": "Apartment",
          "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
          "location": Object {
            "city": "New York",
            "country": "USA",
          },
          "payment": Object {
            "cost": 105,
            "description": "Free cancellation",
          },
          "rating": Object {
            "reviews": 133,
            "stars": 3,
          },
          "title": "Centrally-located Manhattan studio",
        }
      }
    />
    <button
      onClick={[Function]}
    />
  </div>
</li>
`;
